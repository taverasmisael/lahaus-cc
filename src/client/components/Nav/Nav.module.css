.backdrop {
  @apply fixed top-0 left-0 w-full h-full bg-primary;
  @apply transition-opacity duration-100 ease-in-out opacity-0 pointer-events-none;
}

.nav {
  @apply fixed bg-white top-0 left-0 z-50 h-full shadow-xl p-4 grid gap-4;
  @apply -translate-x-full transition-transform duration-300 ease-in-out;
  /* Just to make sure it's not too wide */
  width: clamp(200px, 50%, 370px);
  grid-template-rows: 50px 1fr auto;
}

.navOpen {
  @apply translate-x-0;
}

.backdropOpen {
  @apply opacity-50 pointer-events-auto;
}

.close {
  @apply fixed top-2 right-4 rounded-full w-8 h-8 bg-gray-100 text-primary;
}

.logo {
  @apply inline-block mb-4 w-full h-auto;
}

.menu {
  @apply flex flex-col;
}

.menuItem {
  @apply py-2 text-base font-light hover:text-green-400;
}

.menuItem + .menuItem {
  @apply mt-2;
}

.profileItem {
  @apply text-left flex items-center justify-between relative;
}

.dropdownIcon {
  @apply -rotate-90;
}

.notificationBadge {
  @apply -right-2 top-1;
}

@screen md {
  .backdrop {
    @apply hidden;
  }

  .nav {
    @apply shadow-none relative top-auto left-auto z-0 w-auto flex items-center p-0 transform-none;
  }

  .close {
    @apply hidden;
  }

  .logo {
    @apply hidden;
  }

  .menu {
    @apply block;
  }

  .menuItem {
    @apply px-4;
  }

  .menuItem + .menuItem {
    @apply ml-2;
  }

  .profileItem {
    @apply font-semibold relative;
  }

  .dropdownIcon {
    @apply ml-2 inline-block rotate-0;
  }

  /* Hackish way to get the badge in place */
  .profileItem .notificationBadge {
    @apply block top-1 right-1;
  }
}
