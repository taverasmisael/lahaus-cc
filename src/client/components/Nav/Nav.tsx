import classNames from 'classnames'
import Link from 'next/link'
import NotificationBadge from '@/components/NotificationBadge'

import LogoFull from '../../assets/logo-full.svg'
import DropdownIcon from '../../assets/dropdown-icon.svg'
import styles from './Nav.module.css'

function Nav({ items, hasNotifications, isOpen, onClose }: NavProps) {
  return (
    <>
      <div
        className={classNames(styles.backdrop, {
          [styles.backdropOpen]: isOpen,
        })}
        onClick={onClose}
      />
      <button
        type="button"
        onClick={onClose}
        className={classNames(styles.close, { hidden: !isOpen })}
      >
        <span className="sr-only">Cerrar Men√∫</span>
        <span className={styles.closeIcon} aria-hidden="true">
          &times;
        </span>
      </button>
      <div
        className={classNames('transform', styles.nav, {
          [styles.navOpen]: isOpen,
        })}
      >
        <LogoFull className={styles.logo} title="La haus logo" />

        <nav className={styles.menu}>
          {items.map(({ name, url }) => (
            <Link href={url} key={name}>
              <a className={styles.menuItem}>{name}</a>
            </Link>
          ))}
        </nav>
        <button
          type="button"
          className={classNames(styles.menuItem, styles.profileItem)}
          onClick={() => alert(`I'm not implementing this üòÉ`)}
        >
          Mi Perfil
          <DropdownIcon
            className={classNames('transform', styles.dropdownIcon)}
          />
          {hasNotifications ? (
            <NotificationBadge className={styles.notificationBadge} />
          ) : null}
        </button>
      </div>
    </>
  )
}

interface NavProps {
  items: MenuItem[]
  hasNotifications?: boolean
  isOpen?: boolean
  onClose: () => void
}

interface MenuItem {
  name: string
  url: string
}

export default Nav
