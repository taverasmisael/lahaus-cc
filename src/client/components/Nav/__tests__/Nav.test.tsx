import { render, fireEvent } from '@testing-library/react'
import Nav from '../Nav'

describe('components/Nav', () => {
  test('it can render an empty list', () => {
    const { container } = render(<Nav items={[]} onClose={jest.fn()} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it can render a notification badge', () => {
    const { container } = render(
      <Nav items={[]} hasNotifications onClose={jest.fn()} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('the profile button is present', () => {
    const { getByTestId } = render(<Nav items={[]} onClose={jest.fn()} />)
    getByTestId('profile-button')

    // TODO: test the profile button does something. ATM it only alerts
  })

  test('it can render a list of items', () => {
    const items = [
      { name: 'Item1', url: '#item1' },
      { name: 'Item2', url: '#item2' },
    ]

    const { container } = render(<Nav items={items} onClose={jest.fn()} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it knows how to close', () => {
    const onClose = jest.fn()
    const { getByLabelText } = render(<Nav items={[]} onClose={onClose} />)
    fireEvent.click(getByLabelText(/cerrar men√∫/gi))
    expect(onClose).toHaveBeenCalledTimes(1)
  })
})
