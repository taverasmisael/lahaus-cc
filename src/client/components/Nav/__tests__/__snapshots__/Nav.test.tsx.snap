// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Nav it can render a list of items 1`] = `
<div
  class="backdrop"
/>
`;

exports[`components/Nav it can render a notification badge 1`] = `
<div
  class="backdrop"
/>
`;

exports[`components/Nav it can render an empty list 1`] = `
<div
  class="backdrop"
/>
`;
